src/actions.js

```diff
@@ -0,0 +1,159 @@
+import fetch from 'cross-fetch';
+import { preDelay } from "./constants";
+
+export function signUp(obj) {
+    return dispatch => {
+        dispatch({type: 'START_LOADING'});
+
+        return fetch('/sign-up', {
+            method: "POST",
+            body: obj
+        })
+            .then(response => response.json())
+            .then(data => {
+                if (data.isError){
+                    setTimeout(() => {
```

Comment:
this code needs to be refactored. Create additional functions here and in the same places not to have very looooong functions.